egui_demo_lib/src/apps/demo/paint_bezier.rs

```diff
@@ -0,0 +1,241 @@
+use egui::emath::RectTransform;
+use egui::epaint::{CircleShape, CubicBezierShape, QuadraticBezierShape};
+use egui::*;
+
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[cfg_attr(feature = "serde", serde(default))]
+pub struct PaintBezier {
+    bezier: usize,                       // current bezier curve degree, it can be 3,4,
+    bezier_backup: usize, //track the bezier degree before change in order to clean the remaining points.
+    points: Vec<Pos2>, //points already clicked. once it reaches the 'bezier' degree, it will be pushed into the 'shapes'
+    backup_points: Vec<Pos2>, //track last points set in order to draw auxiliary lines.
+    q_shapes: Vec<QuadraticBezierShape>, //shapes already drawn. once it reaches the 'bezier' degree, it will be pushed into the 'shapes'
+    c_shapes: Vec<CubicBezierShape>, // since `Shape` can't be 'serilized', we can't use Shape as variable type.
```

Comment:
```suggestion
pub struct PaintBezier {
    /// Current bezier curve degree, it can be 3, 4.
    bezier: usize,
    /// Track the bezier degree before change in order to clean the remaining points.
    bezier_backup: usize,
    /// Points already clicked. once it reaches the 'bezier' degree, it will be pushed into the 'shapes'
    points: Vec<Pos2>,
    /// Track last points set in order to draw auxiliary lines.
    backup_points: Vec<Pos2>,
    /// Shapes already drawn. once it reaches the 'bezier' degree, it will be pushed into the 'shapes'
    /// Since `Shape` can't be 'serialized', we can't use Shape as variable type.
    q_shapes: Vec<QuadraticBezierShape>,
    /// Shapes already drawn. once it reaches the 'bezier' degree, it will be pushed into the 'shapes'
    /// Since `Shape` can't be 'serialized', we can't use Shape as variable type.
    c_shapes: Vec<CubicBezierShape>,
```
Also, it seems better to rename `bezier`/`bezier_backup` to `degree`/`degree_backup`

Reply:
thanks, fixed.